File: dsa/week(14)/hasemap.py
from collections import defaultdict

students = {
    "Alice": 'A',
    "Bob": 'B',
    "Charlie": 'A',
    "Diana": 'C',
    "Eve": 'B'
}

# Initialize the gradeGroups dictionary with default values as empty lists
gradeGroups = defaultdict(list)

# Populate the gradeGroups dictionary
for student, grade in students.items():
    gradeGroups[grade].append(student)

# Print the gradeGroups
for grade, group in gradeGroups.items():
    print(f"{grade}: ", end="")
    for student in group:
        print(f"{student}, ", end="")
    print()

File: dsa/week(14)/hasemap.js

const students = {
    "Alice": 'A',
    "Bob": 'B',
    "Charlie": 'A',
    "Diana": 'C',
    "Eve": 'B'
};

// Create an object to hold arrays of students grouped by their grades
const gradeGroups = {};

// Populate the gradeGroups object
for (const [student, grade] of Object.entries(students)) {
    if (!gradeGroups[grade]) {
        gradeGroups[grade] = [];
    }
    gradeGroups[grade].push(student);
}

// Print the grouped students by their grades
for (const [grade, group] of Object.entries(gradeGroups)) {
    process.stdout.write(`${grade}: `);
    for (const student of group) {
        process.stdout.write(`${student}, `);
    }
    console.log();
}
